### PR TITLE
Feat/frontend api integration

### PR DESCRIPTION
**Pull request by SimonSta95**
Added the RAWG API to fetch games.

**Backend:**
- Added needed Models in Backend
- Added Backend endpoints for external API
- Fixed existing endpoints in Backend for API Usage
- Fixed and Added Tests in Backend
- Added Caching to reduce the amount of requests

**Frontend**
- Added needed types in Frontend
- Fixed all components and sites for API usage
- Added Backend search function in Frontend

### CODE REVIEW

File: gamehub-frontend/src/pages/GameGallery/GameGallery.tsx [hunk 1]
```diff
@@ -1,26 +1,89 @@
...
     user: User | null;
-    addGameToLibrary: (gameId: string) => void;
-    deleteGameFromLibrary: (gameId: string) => void;
+    fetchGames: (page: number, searchQuery: string) => void;
+    addGameToLibrary: (game: GameAPI) => void;
+    deleteGameFromLibrary: (game: GameAPI) => void;
 };
 
 export default function GameGallery(props: Readonly<GameGalleryProps>) {
     const [searchQuery, setSearchQuery] = useState('');
```
Comment (rinaehyun) [inline gamehub-frontend/src/pages/GameGallery/GameGallery.tsx:15]: type of searchQuery
`useState<string>('');`

File: gamehub-frontend/src/pages/GameGallery/GameGallery.tsx [hunk 2]
```diff
@@ -1,26 +1,89 @@
...
     const [selectedPlatform, setSelectedPlatform] = useState<string>('');
+    const [currentPage, setCurrentPage] = useState(1);
+    const [loading, setLoading] = useState(false);
```
Comment (rinaehyun) [inline gamehub-frontend/src/pages/GameGallery/GameGallery.tsx:18]: line 17, 18 as well, add type of variable

File: gamehub-frontend/src/pages/GameDetail/GameDetail.tsx [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
-    user: User | null
-}
+    user: User | null;
+};
 
-export default function GameDetail(props: Readonly<GameDetailProps>){
-    const [game, setGame] = useState<Game | undefined | null>(null)
+export default function GameDetail(props: Readonly<GameDetailProps>) {
+    const [game, setGame] = useState<GameDetailAPIResponse | undefined | null>(null);
     const [loading, setLoading] = useState(true);
```
Comment (rinaehyun) [inline gamehub-frontend/src/pages/GameDetail/GameDetail.tsx:14]: `useState<boolean>(true);`

File: gamehub-frontend/src/pages/GameDetail/components/GameNotes.tsx [hunk 4]
```diff
@@ -44,26 +44,24 @@ export default function GameNotes(props: Readonly<GameNotesProps>) {
...
-            created: ""
-        }
-    );
+    const [editNote, setEditNote] = useState<editNote>({
+        title: "",
+        content: "",
+        category: "",
+        created: ""
+    });
     const [confirmDeleteOpen, setConfirmDeleteOpen] = useState(false);
```
Comment (rinaehyun) [inline gamehub-frontend/src/pages/GameDetail/components/GameNotes.tsx:57]: `useState<boolean>(false);`

File: gamehub-frontend/src/pages/GameDetail/components/GameNotes.tsx [hunk 5]
```diff
@@ -44,26 +44,24 @@ export default function GameNotes(props: Readonly<GameNotesProps>) {
     const [newNote, setNewNote] = useState<{ title: string; content: string; category: string }>({
         title: "",
         content: "",
-        category: ""
+        category: "Note" // Default to "Note"
     });
     const [noteLoading, setNoteLoading] = useState(false);
```
Comment (rinaehyun) [inline gamehub-frontend/src/pages/GameDetail/components/GameNotes.tsx:49]: `useState<boolean>(false);`

File: gamehub-frontend/src/types.ts [hunk 6]
```diff
@@ -19,6 +19,57 @@ export type Game = {
...
+export type GameAPI = {
+    id: string,
+    title: string,
+    genre: string[],
+    releaseDate: string,
+    platforms: string[],
+    coverImage: string
+}
+
+export type AddDeleteLibrary = {
```
Comment (rinaehyun) [inline gamehub-frontend/src/types.ts]: The name of type is a little bit confusing. It is for adding library? or deleting library?
Isn't it better to make it more clear, for example AddDeletedLibrary or..? (for your future colleagues :) )

Comment (SimonSta95) [reply gamehub-frontend/src/types.ts]: Thats the fun part. its for both :D It is used while adding to and while deleting from the library

**Review by rinaehyun — APPROVED** [review]
The codes look good!
I just added a few comments. These are only for semantics, not for functional features.

File: gamehub-frontend/src/types.ts (hunk 6 above)
Comment (rinaehyun) [reply gamehub-frontend/src/types.ts]: okay, just asked ChatGPT. It suggested like
- UserGameLibraryAction
- LibraryModification

but it's totally up to you :)

**Pull request merged**